src/proxyfmu/fmi/fmi2/fmi2_slave.hpp

```diff
@@ -21,8 +23,17 @@ class fmi2_slave : public slave
     const model_description md_;
     std::shared_ptr<fmicontext> ctx_;
     std::shared_ptr<temp_dir> tmpDir_;
+    const std::string& instanceName_;
```

Comment:
Safer to use value_type?

Reply:
Can you elaborate a bit on your comment?

Reply:
Since this is a reference to an existing string, the program will invoke undefined behaviour if instanceName no longer exists when used.

using `std::string` makes a copy and omits the issue.

Reply:
Ah you are right, updated.